src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1027,8 +1040,23 @@ public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations
             }
         }
 
-        // PROTOTYPE: need to round trip readonly attribute in metadata
-        internal override bool IsDeclaredReadOnly => false;
+        internal override bool IsDeclaredReadOnly
+        {
+            get
+            {
+                if (!_packedFlags.IsReadOnlyPopulated)
+                {
+                    bool isReadOnly = false;
+                    if (!IsStatic && ContainingType.IsValueType)
```

Comment:
Why aren't we checking for a constructor here? 

Consider centralizing these checks instead of repeating them.

Reply:
Will make an attempt to factor out more of these checks.